### PR TITLE
indexed getter of form elements

### LINKED ISSUE #11405: Implement HTMLFormElement Element getter
This is just a getter that returns the form element at the given index. `HTMLFormControlsCollection.IndexedGetter` should be used.

---
Spec: https://html.spec.whatwg.org/multipage/forms.html#dom-form-item
Code: `components/script/dom/webidl/HTMLFormElement.webidl` `components/script/dom/htmlformelement.rs` `components/script/dom/htmlformcontrolscollection.rs`

### PR DESCRIPTION
**Pull request by mrmiywj**
---

- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [X] These changes fix #11405 (github issue number if applicable).

- [X] There are tests for these changes OR
- [ ] These changes do not require tests because _____

---

This change is [Reviewable](https://reviewable.io/reviews/servo/servo/11697)

### CODE REVIEW

**Comment by Ms2ger** [comment]
You still need to fix the comments made on the previous PR.

**Comment by mrmiywj** [comment]
@Ms2ger 
Hmm, I think I've fixed the comments made on the previous PR.
[This comment](https://reviewable.io/reviews/servo/servo/11657#-KJopXiiN5gn_rzPnwgs:-KJopXij0Iqb7nDVoNjN:705174946) is not avoidable for now.

Other comments have been fixed.

File: components/script/dom/htmlformelement.rs [hunk 1]
```diff
@@ -229,6 +230,12 @@ impl HTMLFormElementMethods for HTMLFormElement {
     fn Length(&self) -> u32 {
         self.Elements().Length() as u32
     }
+
+    // https://html.spec.whatwg.org/multipage/#dom-form-item
+    fn IndexedGetter(&self, index: u32, found: &mut bool) -> Option<Root<Element>> {
+        let elements = self.Elements();
+        (*elements).IndexedGetter(index, found)
```
Comment (Ms2ger) [inline components/script/dom/htmlformelement.rs]: Should be no need for the explicit dereference.

File: tests/wpt/web-platform-tests/html/semantics/forms/the-form-element/form-indexed-element.html [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+<title>form.elements: indexed</title>
+<link rel="author" title="Ivan.Yang" href="mailto:jsyangwenjie@gmail.com">
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<div id="log"></div>
+<div id="test">
+<form id=form>
+<table>
+<tr>
+<td><input type="radio" name="radio1" id="r1" value=1></td>
```
Comment (Ms2ger) [inline tests/wpt/web-platform-tests/html/semantics/forms/the-form-element/form-indexed-element.html]: Why the table?

**Comment by mrmiywj** [comment]
Sorry for that I didn't notice that it was changed on my other PC. My bad.

**Comment by KiChjang** [comment]
Isn't #11405 (the issue that this PR is trying to fix) supposed to be assigned to @catchmrbharath?

**Comment by mrmiywj** [comment]
Sorry for that I did not notice that. This issue was on my ToDo list and still open, so I just tried to pick it up.

File: components/script/dom/webidls/HTMLFormElement.webidl [hunk 3]
```diff
@@ -17,7 +17,7 @@ interface HTMLFormElement : HTMLElement {
 
   [SameObject] readonly attribute HTMLFormControlsCollection elements;
   readonly attribute unsigned long length;
-  //getter Element (unsigned long index);
+  getter Element? (unsigned long index);
```
Comment (KiChjang) [inline components/script/dom/webidls/HTMLFormElement.webidl]: This isn't following the [WebIDL definitions for HTMLFormElement](https://html.spec.whatwg.org/multipage/forms.html#htmlformelement).

File: tests/wpt/web-platform-tests/html/semantics/forms/the-form-element/form-indexed-element.html [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+<link rel="author" title="Ivan.Yang" href="mailto:jsyangwenjie@gmail.com">
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<div id="log"></div>
+<div id="test">
+<form id=form>
+<tr>
+<td><input type="radio" name="radio1" id="r1" value=1></td>
+<td><input type="radio" name="radio2" id="r2" value=2></td>
+</tr>
```
Comment (KiChjang) [inline tests/wpt/web-platform-tests/html/semantics/forms/the-form-element/form-indexed-element.html]: Following on to @Ms2ger's comment, why is this still under a `<tr>` tag?

**Comment by mrmiywj** [comment]
@KiChjang Sorry for that I'm not familiar with FE programming.

**Comment by Ms2ger** [comment]
I'm sorry to do this to you, but @KiChjang was wrong about the `?` in the WebIDL file. In our implementation, it needs to be there.

**Comment by mrmiywj** [comment]
Yep, `?` is not avoided here since the tests failed. Is there any resources about the webidls generator specification used here?

**Comment by jdm** [comment]
https://developer.mozilla.org/en-US/docs/Mozilla/WebIDL_bindings and http://doc.servo.org/script/dom/bindings/index.html might be what you're looking for, or else http://heycam.github.io/webidl/ is the actual specification for WebIDL files.

File: components/script/dom/webidls/HTMLFormElement.webidl (hunk 3 above)
Comment (mrmiywj) [inline components/script/dom/webidls/HTMLFormElement.webidl]: This cannot be avoided for now.

**Comment by KiChjang** [comment]
@bors-servo retry
* The annoying git

**Comment by KiChjang** [comment]
@bors-servo retry
* The super annoying git

**Comment by jdm** [comment]
That test uses indexed getters on form elements, so that's a pleasant surprise :)

**Comment by jdm** [comment]
Please squash the commits :)

**Comment by mrmiywj** [comment]
Sorry, I forgot that;)

**Pull request merged**